sklearn/cluster/tests/test_k_means.py

```diff
@@ -771,4 +774,55 @@ def test_x_squared_norms_init_centroids():
 def test_max_iter_error():
 
     km = KMeans(max_iter=-1)
-    assert_raise_message(ValueError, 'Number of iterations should be', km.fit, X)
+    assert_raise_message(ValueError, 'Number of iterations should be',
+                         km.fit, X)
+
+
+def test_float_precision():
+    km = KMeans(n_init=1, random_state=30)
+    mb_km = MiniBatchKMeans(n_init=1, random_state=30)
+
+    inertia = {}
+    X_new = {}
+    centers = {}
+
+    for estimator in [km, mb_km]:
+        for is_sparse in [False, True]:
+            for dtype in [np.float64, np.float32]:
+                if is_sparse:
+                    X_test = sp.csr_matrix(X_csr, dtype=dtype)
+                else:
+                    X_test = dtype(X)
+                estimator.fit(X_test)
+                # dtype of cluster centers has to be the dtype of the input data
+                assert_equal(estimator.cluster_centers_.dtype, dtype)
+                inertia[dtype] = estimator.inertia_
+                X_new[dtype] = estimator.transform(X_test)
+                centers[dtype] = estimator.cluster_centers_
+                # make sure predictions correspond to the correct label
+                assert_equal(estimator.predict(X_test[0]), estimator.labels_[0])
+                if estimator == mb_km:
```

Comment:
They have the same effect, but different meanings! One is more clear about what you mean to test.